jax/lib/xla_bridge.py

```diff
@@ -179,19 +179,21 @@ def device_count(backend=None):
   return int(get_backend(backend).device_count())
 
 
-def local_device_count(backend=None):
+def local_device_count(backend: str =None):
   """Returns the number of devices on this host."""
   return int(get_backend(backend).local_device_count())
 
 
-def devices(backend=None):
-  """Returns a list of all devices.
+def devices(backend: str = None):
+  """Returns a list of all devices for a given backend.
 
-  Each device is represented by a subclass of Device (e.g. CpuDevice,
-  GpuDevice). The length of the returned list is equal to
-  ``device_count()``. Local devices can be identified by comparing
+  Each device is represented by a subclass of ``Device`` (e.g. ``CpuDevice``,
```

Comment:
A very important concept that is not described here is that of the default device: first device on the default backend. Or is it, first local_device on the default backend?

Reply:
It should be the first local device, but I'm not 100% sure that's true on TPU pods (we use the default XLA device assignment, which in theory can do whatever it wants). We should probably explicitly set a device in jax in the default single-device case, so I'm gonna leave this out for now.